src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,92 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that
+/// a prompt is printed before capturing input.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let name = prompt("Enter name: ");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn prompt(prompt: &str) -> String {
+    stdout().write(prompt.as_bytes()).unwrap();
+    input()
+}
+
+/// Constructs a new handle to the standard input of the current
+/// process, locks this handle and reads a line of input,
+/// returning a `String` containing the input. For automatic
+/// prompt handling, see the [`prompt`] method.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(input)]
+/// use std::io::input;
+///
+/// fn main() {
+///     print!("Enter name: ");
+///
+///     let name = input();
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "input",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn input() -> String {
+    let mut input = String::new();
+    let _ = stdout().flush();
+    let len = stdin().read_line(&mut input).expect("Failed to read from stdin");
+    match len {
+        0 => panic!("invalid character!"),
```

Comment:
```suggestion
        0 => panic!("invalid character"),
```

I feel like it's rude to shout at users.